### PR TITLE
Unittest: AP_Common

### PR DESCRIPTION
**Pull request by khancyr**
Add some more unit test for AP_Common

needs https://github.com/ArduPilot/ardupilot/pull/17720

### CODE REVIEW

**Comment by peterbarker** [comment]
failure-in-failure-path needs fixing:

```
AttributeError: 'CalledProcessError' object has no attribute 'message'
```

**Comment by khancyr** [comment]
@peterbarker rebased, exception handling corrected. Flake8 compliance done.

**Comment by davidbuzz** [comment]
lgtm

File: libraries/AP_Common/tests/test_location.cpp [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    }
+
+//    if (!vehicle.ahrs.set_origin(test_home)) { // TODO: do not work as HAL linux don't update the ins so EKF never initialize
+//        // silently ignored...
+//    }
+}
+
+TEST(Location, NoOrigin)
+{
+    Location test_home(-35362938, 149165085, 100, Location::AltFrame::ABSOLUTE);
```
Comment (peterbarker) [inline libraries/AP_Common/tests/test_location.cpp:204]:
```suggestion
    const Location test_home{-35362938, 149165085, 100, Location::AltFrame::ABSOLUTE};
```

similarly elsewhere

File: libraries/AP_Common/tests/test_location.cpp [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+static GCS_Dummy _gcs;
+AP_Int32 logger_bitmask;
+static AP_Logger logger{logger_bitmask};
+
+const AP_Param::GroupInfo GCS_MAVLINK_Parameters::var_info[] = {
+        AP_GROUPEND
+};
+
+class DummyVehicle : public AP_Vehicle {
```
Comment (peterbarker) [inline libraries/AP_Common/tests/test_location.cpp:21]: Yowsers this is a lot of boiler-plate.  Why do we need a vehicle for this test suite?

**Review by peterbarker — COMMENTED** [review]
Neither of these really needs acting on, but I'm curious to know why we need an entire vehicle when testing location.

If we only need the terrain object - surely we can just supply that?!

File: libraries/AP_Common/tests/test_location.cpp (hunk 2 above)
Comment (khancyr) [reply libraries/AP_Common/tests/test_location.cpp:21]: This is needed to be able to set/get home.
There is surely a better way to do that maybe with a mock class, but I am not used to that yet

**Comment by khancyr** [comment]
@peterbarker that is better now !

**Comment by peterbarker** [comment]
On Thu, 17 Jun 2021, Pierre Kancir wrote:

> This is needed to be able to set/get home.

You only need an AHRS for that.

**Review by peterbarker — APPROVED** [review]

**Pull request merged**